src/test/java/teammates/test/cases/automated/FeedbackSessionUnpublishedMailTest.java

```diff
@@ -0,0 +1,112 @@
+package teammates.test.cases.automated;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.appengine.api.urlfetch.URLFetchServicePb.URLFetchRequest;
+
+import teammates.common.datatransfer.DataBundle;
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.EmailType;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.HttpRequestHelper;
+import teammates.common.util.TimeHelper;
+import teammates.common.util.Const.ParamsNames;
+import teammates.logic.automated.EmailAction;
+import teammates.logic.automated.FeedbackSessionUnpublishedMailAction;
+import teammates.logic.core.CoursesLogic;
+import teammates.logic.core.FeedbackSessionsLogic;
+import teammates.logic.core.InstructorsLogic;
+import teammates.logic.core.StudentsLogic;
+import teammates.test.util.Priority;
+
+@Priority(-1)
+public class FeedbackSessionUnpublishedMailTest extends BaseComponentUsingTaskQueueTestCase {
+    
+    private static final FeedbackSessionsLogic fsLogic = FeedbackSessionsLogic.inst();
+    private static final StudentsLogic studentsLogic = StudentsLogic.inst();
+    private static final InstructorsLogic instructorsLogic = InstructorsLogic.inst();
+    private static final DataBundle dataBundle = getTypicalDataBundle();
+    
+    @SuppressWarnings("serial")
+    public static class FeedbackSessionUnpublishedCallback extends BaseTaskQueueCallback {
```

Comment:
@kanghj `BaseComponentUsingTaskQueueTestCase` is used for `getTypicalDataBundle` and setting up `gaeSimulation` and `assert` statements?